File: .scripts/create.ts
const squareWidth = 512;

function pointToSvgPoint(point: [number, number]): string {
    return `${point[0]},${squareWidth - point[1]}`;
}

const triangleHeight = 256; // The height of the big (not upside-down) triangle.
const smallSquareSideLength = 48;

const circleRadius = squareWidth / 2;

const objectFromTriangleDistance = 10;

const center: [number, number] = [squareWidth / 2, squareWidth / 2];

const triangleTop: [number, number] = [center[0], center[1] + triangleHeight / 2];
console.log(`Found the top of the triangle ${triangleTop}`);

const triangleBottom: [number, number] = [center[0], center[1] - triangleHeight / 2];
console.log(`Found the bottom of the triangle ${triangleBottom}`);

// Calculating the side length s:
// tan(60 degrees) = h / (s / 2)
// (s / 2) * tan(60 degrees) = h
// s = 2h / tan(60 degrees)

const triangleSideLength = (2 * triangleHeight) / Math.tan((60 * Math.PI) / 180);
console.log(`Calculated triangle side length: ${triangleSideLength}`);

const triangleRight: [number, number] = [
    triangleBottom[0] + triangleSideLength / 2,
    triangleBottom[1],
];

const triangleLeft: [number, number] = [
    triangleBottom[0] - triangleSideLength / 2,
    triangleBottom[1],
];

const smallSquareTop: [number, number] = [
    center[0],
    center[1] - Math.sqrt(2) * objectFromTriangleDistance,
];

const smallSquareDiagonal = 2 * Math.sqrt((smallSquareSideLength ** 2) / 2);

const smallSquareCenter: [number, number] = [
    smallSquareTop[0],
    smallSquareTop[1] - smallSquareDiagonal / 2,
];

const smallSquareBottom: [number, number] = [
    smallSquareCenter[0],
    smallSquareCenter[1] - smallSquareDiagonal / 2,
];

const smallSquareRight: [number, number] = [
    smallSquareCenter[0] + smallSquareDiagonal / 2,
    smallSquareCenter[1],
];
const smallSquareLeft: [number, number] = [
    smallSquareCenter[0] - smallSquareDiagonal / 2,
    smallSquareCenter[1],
];

function triangleRightLine(x: number): number {
    return (
        (-triangleHeight / (triangleSideLength / 2)) * (x - squareWidth / 2) +
        squareWidth / 2 +
        triangleHeight / 2
    );
}

const intersectionRight =
    (Math.sqrt(2) * objectFromTriangleDistance +
        (triangleHeight * squareWidth) / triangleSideLength +
        squareWidth / 2 +
        triangleHeight / 2) /
    (1 + triangleHeight / (triangleSideLength / 2));
console.log(`Right intersection: ${intersectionRight}`);

const intersectionLeft =
    squareWidth / 2 - (intersectionRight - squareWidth / 2);
console.log(`Left intersection: ${intersectionLeft}`);

const rightTriangleTopLeft: [number, number] = [
    intersectionRight +
        Math.sqrt(objectFromTriangleDistance ** 2 / 2),
    triangleRightLine(
        intersectionRight +
            Math.sqrt(objectFromTriangleDistance ** 2 / 2)
    ),
];

const leftTriangleTopRight: [number, number] = [
    intersectionLeft -
        Math.sqrt(objectFromTriangleDistance ** 2 / 2),
    rightTriangleTopLeft[0],
];

const intersectionRightBottom =
    (smallSquareDiagonal +
        Math.sqrt(2) * objectFromTriangleDistance +
        (triangleHeight * squareWidth) / triangleSideLength +
        squareWidth / 2 +
        triangleHeight / 2) /
    (1 + triangleHeight / (triangleSideLength / 2));

const intersectionLeftBottom =
    squareWidth / 2 - (intersectionRightBottom - squareWidth / 2);

const rightTriangleBottom: [number, number] = [
    intersectionRightBottom +
        Math.sqrt(objectFromTriangleDistance ** 2 / 2),
    triangleRightLine(
        intersectionRightBottom +
            Math.sqrt(objectFromTriangleDistance ** 2 / 2)
    ),
];

const leftTriangleBottom: [number, number] = [
    intersectionLeftBottom -
        Math.sqrt(objectFromTriangleDistance ** 2 / 2),
    rightTriangleBottom[1],
];

const rightTriangleRight: [number, number] = [
    (smallSquareDiagonal +
        Math.sqrt(2) * objectFromTriangleDistance +
        rightTriangleTopLeft[1]) /
        (1 + triangleHeight / (triangleSideLength / 2)),
    rightTriangleTopLeft[1],
];

let svgString = `<svg width="${squareWidth}" height="${squareWidth}" xmlns="http://www.w3.org/2000/svg">\n`;

svgString += `  <polygon points="${pointToSvgPoint(triangleLeft)} ${pointToSvgPoint(center)} ${pointToSvgPoint(triangleRight)} ${pointToSvgPoint(triangleTop)}" fill="white" />\n`;
svgString += `  <polygon points="${pointToSvgPoint(smallSquareTop)} ${pointToSvgPoint(smallSquareRight)} ${pointToSvgPoint(smallSquareBottom)} ${pointToSvgPoint(smallSquareLeft)}" fill="white" />\n`;
svgString += `  <polygon points="${pointToSvgPoint(rightTriangleBottom)} ${pointToSvgPoint(rightTriangleRight)} ${pointToSvgPoint(rightTriangleTopLeft)}" fill="white" />\n`;

svgString += `</svg>\n`;

console.log(svgString);

File: .scripts/create.py
from math import tan, radians, sqrt

square_width = 512

def point_to_svg_point(point: (float, float)):
    return str((point[0], square_width - point[1])).replace("(", "").replace(")", "")

triangle_height = 256 # The height of the big (not upside down) triangle.
small_square_side_length = 48

circle_radius = square_width / 2

object_from_triangle_distance = 10

center = (square_width / 2, square_width / 2)

triangle_top = (center[0], center[1] + triangle_height / 2) # Point C in the triangle.

print("Found the top of the triangle " + str(triangle_top))

triangle_bottom = (center[0], center[1] - triangle_height / 2) # Not a part of the triangle, instead indicates the bottom line of the triangle.

print("Found the bottom of the triangle " + str(triangle_bottom))

# Calculating the side length s:
# tan(60 degrees) = h/(s/2)
# (s/2) * tan(60 degrees) = h
# s = 2h/tan(60 degrees)

triangle_side_length = 2 * triangle_height / tan(radians(60))

print("Calculated triangle side length: " + str(triangle_side_length))

triangle_right = (triangle_bottom[0] + triangle_side_length / 2, triangle_bottom[1])

triangle_left = (triangle_bottom[0] - triangle_side_length / 2, triangle_bottom[1])

small_square_top = (center[0], center[1] - sqrt(2) * object_from_triangle_distance)

small_square_diagonal = 2 * sqrt((small_square_side_length ** 2) / 2)

small_square_center = (small_square_top[0], small_square_top[1] - small_square_diagonal / 2)

small_square_bottom = (small_square_center[0], small_square_center[1] - small_square_diagonal / 2)

small_square_right = (small_square_center[0] + small_square_diagonal / 2, small_square_center[1])
small_square_left = (small_square_center[0] - small_square_diagonal / 2, small_square_center[1])

def triangle_right_line(x: float):
    return -triangle_height / (triangle_side_length / 2) * (x - square_width / 2) + square_width / 2 + triangle_height / 2

# Line 1: (x - w/2) + w/2 - sqrt(2) * d
# Line 2 (Triangle's right side): -h/(s/2) * (x - w/2) + w/2 + h/2
# -h/(s/2) * (x - w/2) + w/2 + h/2 = x - sqrt(2) * d
# => sqrt(2) * d - h/(s/2) * x + hw/s + w/2 + h/2 = x
# => sqrt(2) * d + hw/s + w/2 + h/2 = x(1 + h/(s/2))
# => (sqrt(2) * d + hw/s + w/2 + h/2)/(1 + h/(s/2)) = x

intersection_right = (sqrt(2) * object_from_triangle_distance + triangle_height * square_width / triangle_side_length + square_width / 2 + triangle_height / 2)/(1 + triangle_height/(triangle_side_length/2))

print("Right intersection: " + str(intersection_right))

intersection_left = square_width / 2 - (intersection_right - square_width / 2)

print("Left intersection: " + str(intersection_left))

right_triangle_top_left = (intersection_right + sqrt(((object_from_triangle_distance) ** 2) / 2), triangle_right_line(intersection_right + sqrt(((object_from_triangle_distance) ** 2) / 2)))

left_triangle_top_right = (intersection_left - sqrt(((object_from_triangle_distance) ** 2) / 2), right_triangle_top_left[0])

# Line 1: (x - w/2) + w/2 - sqrt(2) * d - s_d
# Line 2 (Triangle's right side): -h/(s/2) * (x - w/2) + w/2 + h/2
# -h/(s/2) * (x - w/2) + w/2 + h/2 = x - sqrt(2) * d - s_d
# => s_d + sqrt(2) * d - h/(s/2) * x + hw/s + w/2 + h/2 = x
# => s_d + sqrt(2) * d + hw/s + w/2 + h/2 = x(1 + h/(s/2))
# => (s_d + sqrt(2) * d + hw/s + w/2 + h/2)/(1 + h/(s/2)) = x

intersection_right_bottom = (small_square_diagonal + sqrt(2) * object_from_triangle_distance + triangle_height * square_width / triangle_side_length + square_width / 2 + triangle_height / 2)/(1 + triangle_height/(triangle_side_length/2))

intersection_left_bottom = square_width / 2 - (intersection_right_bottom - square_width / 2)

right_triangle_bottom = (intersection_right_bottom + sqrt(((object_from_triangle_distance) ** 2) / 2), triangle_right_line(intersection_right_bottom + sqrt(((object_from_triangle_distance) ** 2) / 2)))

left_triangle_bottom = (intersection_left_bottom - sqrt(((object_from_triangle_distance) ** 2) / 2), right_triangle_bottom[1])

# Line 1: (x - w/2) + w/2 - sqrt(2) * d - s_d
# Line 2 (Triangle's right side): i_2
# i_2 = x - sqrt(2) * d - s_d
# => s_d + sqrt(2) * d + i_2 = x
# => s_d + sqrt(2) * d + i_2 = x(1 + h/(s/2))
# => (s_d + sqrt(2) * d + i_2)/(1 + h/(s/2)) = x

right_triangle_right = ((small_square_diagonal + sqrt(2) * object_from_triangle_distance + right_triangle_top_left[1])/(1 + triangle_height/(triangle_side_length / 2)), right_triangle_top_left[1])

# SVG CREATION:

svg_string = f"<svg width=\"{square_width}\" height=\"{square_width}\" xmlns=\"http://www.w3.org/2000/svg\">\n"

svg_string += f"  <polygon points=\"{point_to_svg_point(triangle_left)} {point_to_svg_point(center)} {point_to_svg_point(triangle_right)} {point_to_svg_point(triangle_top)}\" fill=\"white\" />\n"
svg_string += f"  <polygon points=\"{point_to_svg_point(small_square_top)} {point_to_svg_point(small_square_right)} {point_to_svg_point(small_square_bottom)} {point_to_svg_point(small_square_left)}\" fill=\"white\" />\n"
svg_string += f"  <polygon points=\"{point_to_svg_point(right_triangle_bottom)} {point_to_svg_point(right_triangle_right)} {point_to_svg_point(right_triangle_top_left)}\" fill=\"white\" />\n"

svg_string += "</svg>\n"

print(svg_string)
